5.0 Draft/IDD/IDDs Authorization System/eu.arrowhead.authorization-management-http-json.yml

```diff
@@ -0,0 +1,518 @@
+openapi: 3.0.3
+info:
+  title: Authorization Policy Management API
+  description: |
+    This specification outlines how to realize the _Arrowhead Authorization Management_ service on top of either of the HTTP or HTTPS
+    protocols with payload encoded in JSON. The service allows for resources, data owners, service providers, or system administrators
+    to manage and express authorization rules to resources. 
+
+    To express authorization policies over resources needing protection, a _resource_ object must be created in the policy engine. This 
+    newly created resource can be _associated_ to existing attributes using the _/management/associations_ endpoint. The same applies
+    whenever a new _subject_ wanting to consume a _resource_ enters to the environment. A _subject_ object must be created in the policy
+    engine, and it needs to be connected to an attribute using the _/management/associations_ endpoint. This association will connect
+    the newly created _resource_ or _subject_ to existing authorization policies. 
+
+
+  version: 1.0.0
+  # Additional information present in the IDD template
+  x-transfer-protocol: [HTTP, HTTPS]
+  x-encryption-method: to-add
+  x-encoding: JSON
+  x-compression: to-add
+  x-semantics: to-add
+  x-Ontology: to-add
+
+paths:
+  /management/associations:
+    post:
+      summary: Create a New association
+      description: Create a new association relating resources to resource-attributes, subjects (consumers) to 
+        subject attributes, or attribute-attribute associations. 
+        The resource or subject must be an existing object. The specified attribute can be one existing from the 
```

Comment:
Is it really necessary for the system providing the authorization service to know about every resource in existence?

Reply:
The service providing the authorization do not need to know about every resource in existence. However, the service managing the policies should know about the existence of the relevant resources when defining policies. For example, when defining a connection between two attributes deriving an access right, the permission should be defined between two attributes that exists (one subject attribute and one resource attribute). In that case, the service that is defining a new policy should be able to know the universe of attributes so that it selects those relevant for the creation of the policies.